.github/workflows/ci.yml

```diff
@@ -15,6 +15,8 @@ on:
   pull_request:
   workflow_dispatch:
 
+permissions: {}
```

Comment:
Including this as a security improvement suggested by `zizmor`.  This will likely conflict with #2975.